### PR TITLE
new TokenList::addAdditionalPostcodeTokens

### PR DESCRIPTION
**Pull request by mtmail**
Logic to find more postcode formats. Core matching logic could probably move to (subclassable?) separate postcode module.

### CODE REVIEW

**Comment by mtmail** [comment]
Found a bug, rebased with master.

Tested against worldwide database and I believe it fixes https://github.com/openstreetmap/Nominatim/issues/1286 and https://github.com/openstreetmap/Nominatim/issues/927

File: lib/TokenList.php [hunk 1]
```diff
@@ -149,6 +149,115 @@ public function addTokensFromDB(&$oDB, &$aTokens, &$aCountryCodes, $sNormQuery,
...
+     *
+     * @param string[] $aTokens       List of tokens to look up in the database.
+     * @param string[] $aCountryCodes List of country restrictions.
+     *
+     * @return void
+     */
+    public function addAdditionalPostcodeTokens(&$aTokens, &$aCountryCodes)
+    {
+        foreach ($aTokens as $sToken) {
+            if ($sToken[0] == ' ') {
```
Comment (lonvia) [inline lib/TokenList.php]: You have left out the check for the token being unmatched: && !$oValidTokens->contains($sToken)

Comment (mtmail) [reply lib/TokenList.php]: added and added a test

**Comment by lonvia** [comment]
Finally closing this. We have a list of postcode regexps now. We just need to integrate them in search somewhat differently.

**Pull request closed without merging**